### DISCUSSION TITLE
Jump / flip issues when trying to fly

### DISCUSSION THREAD

**Discussion by Koraze** [bitcraze/discussions#1391] [Q&A]
Hi, I hope it's the correct channel for my issue.

I got a crazyflie 2.1 with the Flow deck v2 connected last year. I first try to learn how to use my drone with my gamepad and the cfclient on Windows. I already updated the drone firmware to the last clean one. Sadly, my drone crashes every minute.

With "assist mode" on, the flight is quite stable as long as I remain gentle with the controller. However, the drone can sometimes jumps / flips and crash if I make rapid or even sudden movements, particularly on z. Also, if I use the "hover" mode, the drone will crash every time it flies over an object on the ground.

Do you know how to stabilise the drone mouvements by any chance ?

**Comment by gemenerik** [comment]
The flow deck firmware is not designed to fly over uneven terrain. To get a better understanding of the deck check out [this blogpost ](https://www.bitcraze.io/2023/11/go-with-the-flow-relative-positioning-with-the-flow-deck/). Please try flying over a flat surface and let me know how it goes!

**Reply by Itai51** [reply]
Hello, I am having a similar issue when following the [Motion Commander tutorial, step 2](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_motion_commander/#step-2-take-off-function), where after 1-1.5 seconds of a shaky takeoff the CF flips over and crashes (often with the propellors still spinning into the floor).

I read the link in your comment, and I have addressed all of the points:
- I have a wooden floor with natural texture
- the floor is not too shiny
- the room is well-lit
- I closed the blinds to block sunlight

I even replaced the propellors and one of the motors arms. Here is a picture of my environment:
![WhatsApp Image 2024-06-24 at 12 04 32_f7baf355](https://github.com/bitcraze/discussions/assets/42220654/ebee531d-dd94-442d-bdbd-25846a295634)

What else can I do?

Thank you!

**Reply by knmcguire** [reply]
Hmmm.. this is strange! This would be a perfect surface for the Crazyflie's flowdeck to fly over. I feel that there is something else going on.

Could you perhaps test the crazyflie out by hand, and look at the plotting to see the variables motion.x, motion.y. Are they able to see any flow at all?

Also, do you see any error in the console tab as well?

**Reply by knmcguire** [reply]
Oh wait, sorry I saw that you were not the original poster. Let's talk further in here to keep it separate since your issue could be different: https://github.com/orgs/bitcraze/discussions/1417

**Reply by Itai51** [reply]
I ended up recalibrating my LHs, because the last time I calibrated was on the table, so the floor was about z = -0.74m.

I am now flying stable!

**Reply by knmcguire** [reply]
ah oke. So you were not using a flowdeck then like the original poster.

In any case it is usually easier for us to treat your case as an new issue if you start new discussion thread. If you continue on some-one else's, then we assume that you are dealing with the exact same problem with the exact same hardware. Just a pointer for next time.

**Reply by Koraze** [reply]
Hi again, sorry for my late reply

I didn't understand that the flow deck shield was not designed to fly only on perfectly smooth terrain. This is very sad news for me.

I wanted to use bitcraze drones both as a simple programmable drone (preferably in python) for beginners, and as a demonstrator for simple projects. Unfortunately, I can't always control the quality of the environment. So I'll look for another drone and keep an eye on future developments of your drones.

Sorry to ask you this, but do you know another drone (if possible open-source) that could meets my needs?

**Reply by knmcguire** [reply]
As long as there is some natural texture (like the picture that another poster did, it should always work: I've also kept painterstape with me to create some kind of texture adhoc but unfortunately that is just how it goes with optical flow. No texture, no dice, and that unfortunately holds for anything else that works on fast sparse optical flow.

You could check out the dji tello but that got discontinued and is also not opensource. There are other options here but they increase quite a lot in price: https://ros-aerial.github.io/aerial_robotic_landscape/hardware/

**Reply by knmcguire** [reply]
But the jumps you are seeing don't have anything to do with the flow so if you have a picture of your environment we can help out a bit. It maybe has something to do with your height sensor
